server/fleet/hosts.go

```diff
@@ -540,6 +540,7 @@ func (h *Host) IsElegibleForDEPMigration() bool {
 // fleet but it's currently unenrolled.
 func (h *Host) NeedsDEPEnrollment() bool {
 	return !h.MDMInfo.IsDEPFleetEnrolled() &&
+		!h.MDMInfo.IsManualFleetEnrolled() &&
```

Comment:
Why would a device that is manually enrolled to Fleet also be DEP assigned to Fleet?

Reply:
They can still install an enrollment profile manually for whatever reason 😓 , if that's the case, we don't want to show the migration dialog.